tests/Console/Command/StageCommandTest.php

```diff
@@ -30,22 +47,88 @@ public function testBasicConfiguration(): void
         $definition = $command->getDefinition();
         $arguments = $definition->getArguments();
         $options = $definition->getOptions();
+        $composerCommandArgument = $definition->getArgument('composer-command');
 
         self::assertSame('stage', $command->getName(), 'Set correct name.');
         self::assertSame([], $command->getAliases(), 'Set correct aliases.');
         self::assertNotEmpty($command->getDescription(), 'Set a description.');
-        self::assertSame([], array_keys($arguments), 'Set correct arguments.');
+        self::assertSame(['composer-command'], array_keys($arguments), 'Set correct arguments.');
         self::assertSame([], array_keys($options), 'Set correct options.');
+        self::assertNotEmpty($command->getUsages(), 'Set usages.');
+        self::assertNotEmpty($command->getHelp(), 'Set help.');
+
+        self::assertTrue($composerCommandArgument->isRequired(), 'Required Composer command option.');
+        self::assertTrue($composerCommandArgument->isArray(), 'Required Composer command option.');
+        self::assertNotEmpty($composerCommandArgument->getDescription(), "Description provided.");
     }
 
     /**
      * @covers ::execute
+     *
+     * @dataProvider providerBasicExecution
      */
-    public function testBasicExecution(): void
+    public function testBasicExecution($composerCommand, $stagingDir, $output): void
     {
-        $this->executeCommand();
+        $this->stager
+            ->stage($composerCommand, $stagingDir)
+            ->shouldBeCalledOnce()
+            ->willReturn($output);
+
+        $this->executeCommand([
+            'composer-command' => $composerCommand,
+            '--staging-dir' => $stagingDir,
+        ]);
 
-        self::assertSame('', $this->getDisplay(), 'Displayed correct output.');
+        self::assertSame($output, $this->getDisplay(), 'Displayed correct output.');
         self::assertSame(ExitCode::SUCCESS, $this->getStatusCode(), 'Returned correct status code.');
     }
+
+    public function providerBasicExecution(): array
+    {
+        return [
```

Comment:
I'm not entirely clear on what these test cases prove -- is it just showing that the command calls the stager? If so, it'd be nice if the test method documented that.

Reply:
Yes, that it calls the stager with the correct arguments and that it emits the output it gets back from it and returns the correct status code. I always do a happy path test. It's not always very interesting. 🙂

Reply:
Makes sense. A doc comment would then be useful :)

Reply:
I'm actually going to leave this, because I use the same pattern just _everywhere_. Explaining it in just one instance would be arbitrary, and explaining it in every one would be tedious and wasteful.